dht_net.go

```diff
@@ -133,13 +164,10 @@ type messageSender struct {
 	p   peer.ID
 	dht *IpfsDHT
 
+	invalid   bool
```

Comment:
what is the purpose of this? doesn't seem to be used.

Reply:
Now it is. We were toying with using it and decided we didn't need it but forgot to remove it. Now, we realized we really do need it.

---

If you can think of a better, race-free way of doing this dance, suggestions welcome!